### ISSUE TITLE
Conversation queues should be replaced by a statemachine

### ISSUE THREAD

**Issue by ashneo76** [ramswaroop/jbot#19]
Implementing the conversation as a queue can lead to stack overflows when conversation needs to be restarted from a point already mentioned in a method before. An acyclic directed graph can be used to handle such a situation.

**Comment by ramswaroop** [comment]
Yes, I need to think on this. There is another issue opened which is related to this: https://github.com/ramswaroop/jbot/issues/13
